contracts/src/libraries/verifier/ZkTrieVerifier.sol

```diff
@@ -58,19 +56,20 @@ library ZkTrieVerifier {
                 }
             }
             // compute poseidon hash of two uint256
-            function poseidon_hash(hasher, v0, v1) -> r {
+            function poseidon_hash(hasher, v0, v1, domain) -> r {
                 let x := mload(0x40)
-                // keccack256("poseidon(uint256[2])")
-                mstore(x, 0x29a5f2f600000000000000000000000000000000000000000000000000000000)
+                // keccack256("poseidon(uint256[2],uint256)")
+                mstore(x, 0xa717016c00000000000000000000000000000000000000000000000000000000)
```

Comment:
where is this magic number from? the entry of function selector?

Reply:
yes